lib/active_merchant/billing/gateways/cyber_source_rest.rb

```diff
@@ -47,6 +47,19 @@ def authorize(money, payment, options = {}, capture = false)
         commit('/pts/v2/payments/', post)
       end
 
+      def void(authorization, options = {})
+        post = { request: { OrderId: authorization } }
+        commit("/pts/v2/payments/#{authorization}/voids", post)
+      end
+
+      def verify(credit_card, options = {})
+        amount = eligible_for_zero_auth?(credit_card, options) ? 0 : 100
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { purchase(amount, credit_card, options) }
```

Comment:
Some special reason to use purchase instead Authorize ❓

Reply:
I had been wrong doing a **void purchase** but in reality I needed to do a **reversal**

In the [documentation](https://developer.cybersource.com/api-reference-assets/index.html#payments_reversal_process-an-authorization-reversal_samplerequests-dropdown_process-an-authorization-reversal_liveconsole-tab-request-body) void is used for a cancellation and reverse for a void (in the AM context)

Reply:
Then we change the approach and now was implemented using **authorize**